docs/root/start/sandboxes/single-page-app.rst

```diff
@@ -0,0 +1,648 @@
+
+.. _install_sandboxes_single_page_app_step_login:
+
+Step 4: Browse to the dev app and login
+***************************************
+
+The development app should now be available at http://localhost:10001 and provide a login button:
+
+.. image:: /start/sandboxes/_static/spa-login.png
+   :align: center
+
+.. note::
+   The dummy OAuth provider automatically trusts everyone as a hard-coded ``envoydemo`` user and redirects back to the app.
+
+   In a real world scenario the provider would authenticate and authorize the user before proceeding.
+
+The sandbox is configured with an inverted match on
+:ref:`pass_through_matcher <envoy_v3_api_field_extensions.filters.http.oauth2.v3.OAuth2Config.pass_through_matcher>`.
+
+This ignores all paths for OAuth other than:
+
+- ``/authorize.*``
+- ``/hub.*``
+- ``/login``
+- ``/logout``.
+
+.. literalinclude:: _include/single-page-app/envoy.yml
+   :language: yaml
+   :lines: 37-46
+   :linenos:
+   :lineno-start: 37
+   :emphasize-lines: 3-8
+   :caption: :download:`envoy.yml <_include/single-page-app/envoy.yml>`
+
+When a user clicks ``login`` the app initiates the OAuth flow by calling the ``/login`` path in Envoy.
+
+This redirects the user to the OAuth provider for authorization/authentication with a further redirect link:
+
+.. literalinclude:: _include/single-page-app/envoy.yml
+   :language: yaml
+   :lines: 34-39
+   :linenos:
+   :lineno-start: 34
+   :emphasize-lines: 3-4
+   :caption: :download:`envoy.yml <_include/single-page-app/envoy.yml>`
+
+On successful authorization/authentication the user is redirected back via this link to the app with the necessary OAuth
+`authorization code <oauth-auth-code_>`__ to proceed:
+
+.. literalinclude:: _include/single-page-app/envoy.yml
+   :language: yaml
+   :lines: 44-50
+   :linenos:
+   :lineno-start: 44
+   :emphasize-lines: 3-5
+   :caption: :download:`envoy.yml <_include/single-page-app/envoy.yml>`
+
+Envoy then uses this authorization code with its client secret to confirm authorization and obtain an access token for the user:
+
+.. literalinclude:: _include/single-page-app/envoy.yml
+   :language: yaml
+   :lines: 50-61
+   :linenos:
+   :lineno-start: 50
+   :emphasize-lines: 3-10
+   :caption: :download:`envoy.yml <_include/single-page-app/envoy.yml>`
+
+.. literalinclude:: _include/single-page-app/secrets/myhub-token-secret.yml
+   :language: yaml
+   :linenos:
+   :emphasize-lines: 6
+   :caption: :download:`myhub-token-secret.yml <_include/single-page-app/secrets/myhub-token-secret.yml>`
+
+Once logged in, you should be able to make queries to the API using the OAuth credentials:
+
+.. image:: /start/sandboxes/_static/spa-resources.png
+   :align: center
+
+.. warning::
+   Envoy's OAuth implementation defaults to triggering the OAuth flow for all paths on the endpoint.
+
+   This can readily trigger an OAuth flood as assets are requested, and doom loops when the OAuth flows fail.
```

Comment:
Not sure I fully follow - shouldn't the access token creation obtain sufficient scope to be able to reuse the OAuth cookie in future accesses to other paths?

Reply:
if you dont restrict what triggers the oauth flow then its triggered on the assets

this can cause redirect loops which take over the client (browser) machine and flood the proxy

not knowing this when i started i really struggled to get anything working so i think we need to say something - here im trying to highlight the strategy/config used to avoid that (ie inverse matching)